### PR TITLE
Fix GitLab jobs' Id in mocks

### PR DESCRIPTION
**Pull request by jairbubbles**

### CODE REVIEW

File: NGitLab.Mock.Tests/ConfigTests.cs [hunk 1]
```diff
@@ -16,24 +18,17 @@ public void Test_server_can_be_saved_in_config()
             server.Groups.Add(group);
             var project = new Project("test-project")
             {
-                Description = "Test project",
-                DefaultBranch = "default",
-                Visibility = VisibilityLevel.Public,
+                Description = "Test project", DefaultBranch = "default", Visibility = VisibilityLevel.Public,
```
Comment (louis-z) [inline NGitLab.Mock.Tests/ConfigTests.cs]: Why set properties on a single line? IMO leaving them spread over multiple lines makes it more readable.

File: NGitLab.Mock.Tests/ConfigTests.cs [hunk 2]
```diff
@@ -107,5 +102,33 @@ public void Test_config_can_be_serialized()
             using var server = config2.BuildServer();
             Assert.IsNotNull(server);
         }
+
+        [Test]
+        public void Test_jobs_id_are_unique()
```
Comment (louis-z) [inline NGitLab.Mock.Tests/ConfigTests.cs]:
```suggestion
        public void Test_job_ids_are_unique()
```

File: NGitLab.Mock/Config/GitLabHelpers.cs [hunk 3]
```diff
@@ -947,6 +947,9 @@ public static GitLabPipeline WithJob(this GitLabPipeline pipeline, string? name
         {
             return Configure(pipeline, _ =>
             {
+                if (pipeline.Parent == null)
+                    throw new InvalidOperationException($"Parent not set on pipeline {pipeline}");
```
Comment (louis-z) [inline NGitLab.Mock/Config/GitLabHelpers.cs]:
```suggestion
                    throw new InvalidOperationException($"Parent project not set for pipeline {pipeline}");
```

File: NGitLab.Mock.Tests/NGitLab.Mock.Tests.csproj [hunk 4]
```diff
@@ -10,5 +10,6 @@
     <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.4.1" />
     <PackageReference Include="NUnit" Version="3.13.3" />
     <PackageReference Include="NUnit3TestAdapter" Version="4.3.1" />
+    <PackageReference Include="FluentAssertions" Version="6.7.0" />
```
Comment (Toa741) [inline NGitLab.Mock.Tests/NGitLab.Mock.Tests.csproj:13]: To the other owners (@louis-z , @meziantou), are we ok to add FluentAssertions?

File: NGitLab.Mock.Tests/ConfigTests.cs (hunk 1 above)
Comment (jairbubbles) [reply NGitLab.Mock.Tests/ConfigTests.cs]: I'll revert, it's auto refactoring's fault.

File: NGitLab.Mock.Tests/NGitLab.Mock.Tests.csproj (hunk 4 above)
Comment (louis-z) [reply NGitLab.Mock.Tests/NGitLab.Mock.Tests.csproj:13]: I don't have any objections.

**Pull request merged**